dev-tools.asciidoc

```diff
@@ -367,13 +367,30 @@ Website link; https://openzeppelin.org/
 
 Read the docs link; http://zeppelin-solidity.readthedocs.io/en/latest/index.html
 
-=== ZeppelinOS
-zeppelin_os is an open-source, distributed platform of tools and services on top of the EVM
-to develop and manage smart contract applications securely.
+=== zeppelin_os
+https://github.com/zeppelinos[zeppelin_os] is an open-source, distributed platform of tools and services on top of the EVM
+to develop and manage smart contract applications securely. 
 
-Github link; https://github.com/zeppelinos
+Unlike OpenZeppelin's code, which needs to be re-deployed with each application every time, zeppelin_os's code lives on-chain. Applications that need a given functionality, say, an ERC20 token, not only don't have to re-design and re-audit its implementation (something that OpenZeppelin solved) but don't even need to deploy it. With zeppelin_os, an application interacts with the token's on-chain implementation directly, much in the same way a desktop application interacts with the components of its underlaying OS.
+
+Applications that make use of OpenZeppelin avoid having to "reinvent the wheel" by reusing the library's curated and peer reviewed contracts. However, every single time an application uses an ERC20 token implementation, the same ERC20 bytecode is deployed into the blockchain, over and over again. Such bytecode per-se, exists duplicated in the network countless times. Now, applications that make use of zeppelin_os avoid this unnecessary duplication. Instead of deploying their own ERC20 implementation, they link to a contract that defines the latest ERC20 implementation accepted by the community. This single, central implementation is deployed in the blockchain only once, much like Solidity's libraries, but are considerably more sophisticated.
+
+Unlike Solidity's libraries, the implementations offered by zeppelin_os can be treated like regular contracts and, are upgradeable. If ever a vulnerability is found in one of the OS's official implementations, it can simply be swapped with an upgraded one. In the case of the ERC20 token, an upgrade to its implementation would instantly ripple to all the applications that use it. The OS not only provides upgradeability for all of its implementations, but also for the user's own contracts, and even for it's own codebase! Developers decide when and how an application should implement an upgrade, and even which "flavor" of an implementation to adhere to. 
```

Comment:
What does "can be treated like regular contracts" mean? Seems like it links with following "and are upgradeable", but regular contracts are not upgradeable.

Reply:
Right. Will try to clear it out a bit.